synapse/models/material.py

```diff
@@ -31,37 +31,88 @@ class MatModule(s_module.CoreModule):
 
     def getModelDefs(self):
         modl = {
+
+            'interfaces': (
+
+                ('phys:object', {
+                    'doc': 'Properties common to all physical objects.',
+                    'template': {'phys:object': 'object'},
+                    'props': (
+
+                        ('phys:mass', ('mass', {}), {
+                            'doc': 'The mass of the {phys:object}.'}),
+
+                        ('phys:volume', ('geo:dist', {}), {
+                            'doc': 'The cubed volume of the {phys:object}.'}),
+
+                        ('phys:length', ('geo:dist', {}), {
+                            'doc': 'The length of the {phys:object}.'}),
+
+                        ('phys:width', ('geo:dist', {}), {
+                            'doc': 'The width of the {phys:object}.'}),
+
+                        ('phys:height', ('geo:dist', {}), {
+                            'doc': 'The height of the {phys:object}.'}),
+                    ),
+                }),
+            ),
+
             'types': (
-                ('mat:item', ('guid', {}), {'doc': 'A GUID assigned to a material object.'}),
+
+                ('phys:object', ('ndef', {'interface': 'phys:object'}), {
+                    'doc': 'A node which represents a physical object.'}),
+
+                ('phys:contained', ('guid', {}), {
+                    'doc': 'A node which represents a physical object containing another physical object.'}),
+
+                ('mat:item', ('guid', {}), {
+                    'interfaces': ('phys:object', 'geo:locatable'),
+                    'template': {'phys:object': 'item', 'geo:locatable': 'item'},
+                    'doc': 'A GUID assigned to a material object.'}),
+
                 ('mat:type', ('taxonomy', {}), {
                     'doc': 'A taxonomy of material item/specification types.',
-                    'interfaces': ('meta:taxonomy',),
-                }),
+                    'interfaces': ('meta:taxonomy',)}),
+
                 ('mat:spec', ('guid', {}), {'doc': 'A GUID assigned to a material specification.'}),
                 ('mat:specimage', ('comp', {'fields': (('spec', 'mat:spec'), ('file', 'file:bytes'))}), {}),
                 ('mat:itemimage', ('comp', {'fields': (('item', 'mat:item'), ('file', 'file:bytes'))}), {}),
 
                 ('mass', ('hugenum', {'units': massunits}), {
                     'doc': 'A mass which converts to grams as a base unit.'}),
-                # TODO add base type for volume
             ),
 
             'forms': (
 
+                ('phys:contained', {}, (
```

Comment:
Something about this is throwing me a little. Because we're using "contained" (vs "contains") it feels like we are emphasizing the object being contained, as opposed to the object that is the container (but also a phys:object) itself. (So focusing on thing 1 that is within thing 2, as opposed to capturing it as thing 2 which contains thing 1). I think this is bothering me because I interpret the container as the primary object and phys:contains would communicate that focus (and be consistent with the phrasing "the period where the container held the object"). However, I'm not at all sure that this actually matters

Reply:
The node represents the time period where the one object contained the other ( both are separate physical objects ) but def open to discussing the name of the form :+1: